lib/fizzy/execute.cpp

```diff
@@ -418,13 +419,14 @@ template <typename T>
 inline T shift_right(T lhs, T rhs) noexcept
 {
     constexpr T num_bits{sizeof(T) * 8};
-    const auto k = rhs % num_bits;
+    const auto k = rhs & (num_bits - 1);
```

Comment:
Just an idea, but might be more readable.
```suggestion
    constexpr T mask{sizeof(T) * 8 - 1};
    const auto k = rhs & mask;
```

Reply:
not sure it is